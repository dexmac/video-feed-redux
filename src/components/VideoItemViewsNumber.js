import React from "react";
import PropTypes from "prop-types";

const VideoItemViewsNumber = ({ viewsNumber }) => (
  <div className="video-item-views-number">
    {viewsNumber >= 0 ? "Views: " + viewsNumber : ""}
  </div>
);

VideoItemViewsNumber.propTypes = {
  viewsNumber: PropTypes.number
};

export default VideoItemViewsNumber;
