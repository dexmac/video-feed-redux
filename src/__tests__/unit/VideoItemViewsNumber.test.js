import React from "react";
import VideoItemViewsNumber from "../../components/VideoItemViewsNumber";

describe("VideoItemViewsNumber", () => {
  let props;
  let mountedVideoItemViewsNumber;

  /**
   * Clearing props before each test
   */
  beforeEach(() => {
    props = {
      viewsNumber: undefined
    };
    mountedVideoItemViewsNumber = undefined;
  });

  /**
   * Shallow rendering vs. snapshot test
   */
  it("renders correctly && matches snapshot", () => {
    const tree = shallow(<VideoItemViewsNumber />);
    expect(tree).toMatchSnapshot();
  });

  /**
   * Mounts the component using Enzyme if it is not mounted yet
   * @returns {object} - returns the mounted component
   */
  const videoItemViewsNumber = () => {
    if (!mountedVideoItemViewsNumber) {
      mountedVideoItemViewsNumber = mount(<VideoItemViewsNumber {...props} />);
    }
    return mountedVideoItemViewsNumber;
  };

  describe("when `viewsNumber` is defined", () => {
    beforeEach(() => {
      props.viewsNumber = 123123;
    });

    it("sets the rendered `div` text to the same value as `viewsNumber`", () => {
      const myDiv = videoItemViewsNumber().find("div");
      const regExpToFindViewsNumber = new RegExp(
        props.viewsNumber ? props.viewsNumber : null,
        "g"
      );
      expect(myDiv.text()).toMatch(regExpToFindViewsNumber);
    });
  });

  describe("when `viewsNumber` is undefined", () => {
    beforeEach(() => {
      props.viewsNumber = undefined;
    });

    it("sets the rendered `div` text to be an empty string", () => {
      const myDiv = videoItemViewsNumber().find("div");
      expect(myDiv.text()).toBe("");
    });
  });
});
