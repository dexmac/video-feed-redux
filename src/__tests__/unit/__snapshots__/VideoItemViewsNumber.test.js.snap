// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoItemViewsNumber renders correctly && matches snapshot 1`] = `
<div
  className="video-item-views-number"
/>
`;
